internal/goofys.go

```diff
@@ -96,61 +96,65 @@ var log = GetLogger("main")
 var fuseLog = GetLogger("fuse")
 
 func NewBackend(bucket string, flags *FlagStorage) (cloud StorageBackend, err error) {
-	if flags.Backend == nil {
-		flags.Backend = (&S3Config{}).Init()
-	}
-
-	if config, ok := flags.Backend.(*AZBlobConfig); ok {
+	if config, ok := flags.BackendConfig.(*AZBlobConfig); ok {
 		cloud, err = NewAZBlob(bucket, config)
-	} else if config, ok := flags.Backend.(*ADLv1Config); ok {
+	} else if config, ok := flags.BackendConfig.(*ADLv1Config); ok {
 		cloud, err = NewADLv1(bucket, flags, config)
-	} else if config, ok := flags.Backend.(*S3Config); ok {
+	} else if config, ok := flags.BackendConfig.(*S3Config); ok {
 		if strings.HasSuffix(flags.Endpoint, "/storage.googleapis.com") {
 			cloud = NewGCS3(bucket, flags, config)
 		} else {
 			cloud = NewS3(bucket, flags, config)
 		}
 	} else {
-		err = fmt.Errorf("Unknown backend config: %T", flags.Backend)
+		err = fmt.Errorf("Unknown backend config: %T", flags.BackendConfig)
 	}
 
 	return
 }
 
-type BucketSpec struct {
-	Scheme string
-	Bucket string
-	Prefix string
-}
-
-func ParseBucketSpec(bucket string) (spec BucketSpec, err error) {
-	if strings.Index(bucket, "://") != -1 {
-		var u *url.URL
-		u, err = url.Parse(bucket)
+// FillBackendConfig creates config backend in the flags due to
+// the backend-storage flag
+func FillBackendConfig(bucketName string, flags *FlagStorage) error {
+	switch flags.Backend {
+	case "azure-data-lake":
```

Comment:
change this to `azure-data-lake-v1`, there's also `azure-data-lake-v2` that's going to be added later

Reply:
Okay